linq4j/src/main/java/org/apache/calcite/linq4j/tree/Expressions.java

```diff
@@ -1507,9 +1507,6 @@ public static TernaryExpression makeTernary(ExpressionType ternaryType,
     case Conditional:
       if (e1 instanceof ConstantUntypedNull) {
         type = Primitive.box(e2.getType());
-        if (e1.getType() != type) {
```

Comment:
Why is this "if" block removed, but not the one just below (lines 1515..1517), which is the same but with `e2` instead of `e1`? Could that one be removed too?

Reply:
thanks, seems you are right, i change it, will wait new test run completed.

Reply:
I force push suggested change and it`s all ok from CI side.